src/librustc_mir_build/build/expr/as_constant.rs

```diff
@@ -21,7 +21,8 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         let Expr { ty, temp_lifetime: _, span, kind } = expr;
         match kind {
             ExprKind::Scope { region_scope: _, lint_level: _, value } => this.as_constant(value),
-            ExprKind::Literal { literal, user_ty } => {
+            ExprKind::Literal { literal, user_ty } |
+            ExprKind::StaticRef { literal, user_ty, .. } => {
```

Comment:
Here and below we have more strange changes that seem entirely unrelated to the PR description.